spec/models/course_module_spec.rb

```diff
@@ -25,4 +25,26 @@
     it { is_expected.to validate_presence_of(:title).with_message("Enter a title") }
     it { is_expected.to validate_presence_of(:content).with_message("Enter content") }
   end
+
+  describe "course_lessons" do
+    it "returns lessons in order after they get updated" do
+      course_module = FactoryBot.create(:course_module)
+      course_lesson_one = FactoryBot.create(:course_lesson, title: "One", course_module: course_module)
+      course_lesson_two = FactoryBot.create(:course_lesson, title: "Two", course_module: course_module)
+      course_lesson_three = FactoryBot.create(:course_lesson, title: "Three", course_module: course_module)
+      course_lesson_four = FactoryBot.create(:course_lesson, title: "Four", course_module: course_module)
+
+      course_lesson_two.previous_lesson = course_lesson_one
+      course_lesson_four.previous_lesson = course_lesson_two
+      course_lesson_three.previous_lesson = course_lesson_four
+      course_lesson_two.save!
+      course_lesson_three.save!
+      course_lesson_four.save!
+
+      expected_lessons_with_order = [course_lesson_one, course_lesson_two, course_lesson_four, course_lesson_three]
+      expected_lessons_with_order.zip(course_module.course_lessons).each do |expected, actual|
+        expect(expected).to eq(actual)
```

Comment:
Hmm reads kind of funny - expect the expected and you never be disappointed - if this was a story and you would read it as a bedtime story - how could you improve the plotline ?